### PR TITLE
feat(api): standardize & document error responses to rfc7807

### PR DESCRIPTION
**Pull request by Goncharo**
refs. metriport/metriport#259

### Dependencies

N/A

### Description

standardize & document error responses to rfc7807 - docs preview:

Screenshot 2023-04-28 at 8 08 56 PM

### Release Plan

ASAP, nothing special

### CODE REVIEW

**Review by Orta21 — APPROVED** [review]

File: docs/home/api-info/error-responses.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+</Tip>
+
+## Format
+
+```json
+{
+  "status": 500,
+  "name": "INTERNAL_SERVER_ERROR",
+  "title": "NotFoundError",
+  "detail": "Could not find organization Devault Medical"
```
Comment (leite08) [inline docs/home/api-info/error-responses.mdx:25]: I suggest we make those related, maybe status to `404` and name to `NOT_FOUND`.

File: docs/home/api-info/error-responses.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<ResponseField name="name" type="string" required>
+  Human-readable description of the status code.
+</ResponseField>
+
+<ResponseField name="title" type="string" required>
+  The specific error description - this wont't change between occurences of the same error.
+</ResponseField>
+
+<ResponseField name="detail" type="string" required>
+  Details about this error occurrence.
```
Comment (leite08) [inline docs/home/api-info/error-responses.mdx:42]: Maybe add: ` - this might change for each instance/request.`

**Review by leite08 — CHANGES_REQUESTED** [review]
Looks good, but left a few messages that I thing would be nice to review (most optional).

Also, if you could update `ddbItemAsOAuth`, the message there is sending the token on the error message.

(optional) This is not part of this PR, but there are a few instances where we throw a new `Error` on a `catch` without sending the original error as `cause` - and also not sending the original error to Sentry, only to logs. This makes it harder to debug. Sharing in case you want to use this PR to address that too - search for `throw new Error(msg)`, `throw new Error("Google`, `throw new Error(`Request failed`.

File: api/app/src/default-error-handler.ts [hunk 3]
```diff
@@ -43,21 +49,25 @@ export const errorHandler: ErrorRequestHandler = (err, req, res, next) => {
...
+    .send({
+      ...defaultResponseBody({
         status: status.INTERNAL_SERVER_ERROR,
-        title: "Internal server error, please try again or reach out to support@metriport.com",
-      })
-    );
+        title: "InternalServerError",
+        detail: "Please try again or reach out to support@metriport.com",
+      }),
+      name: status[status.INTERNAL_SERVER_ERROR],
```
[the remark this answers is not in the document]
Comment (Goncharo) [reply api/app/src/default-error-handler.ts:71]: What's the logic behind putting these in a `const` being better?

They're only used once in one place, and map lookup is O(1)

[the remark this answers is not in the document]
Comment (leite08) [reply api/app/src/default-error-handler.ts:71]: So we dont define the status code 3 times - not concerned with map access. 😄

**Review by leite08 — APPROVED** [review]

**Pull request merged**